pkg/diagnostics/diagnostic_bundle.go

```diff
@@ -72,9 +72,12 @@ func newDiagnosticBundleManagementCluster(af AnalyzerFactory, cf CollectorFactor
 	b.WithDefaultCollectors().
 		WithDefaultAnalyzers().
 		WithManagementCluster(true).
-		WithDatacenterConfig(spec.Cluster.Spec.DatacenterRef).
-		WithAPIServerCollectors(spec.Cluster.Spec.ControlPlaneConfiguration.Endpoint.Host). // Add a collector here to runPod and ping control plane IP once bootstrap cluster was created
-		WithLogTextAnalyzers()
+		WithDatacenterConfig(spec.Cluster.Spec.DatacenterRef)
+
+	if spec.Cluster.Spec.ControlPlaneConfiguration.Endpoint != nil && spec.Cluster.Spec.ControlPlaneConfiguration.Endpoint.Host != "" {
+		b.WithAPIServerCollectors(spec.Cluster.Spec.ControlPlaneConfiguration.Endpoint.Host)
```

Comment:
im not familiar w this api server collector -- does it make sense that if no cp endpoint host specified, we don't check the ap server?

Reply:
If aim to create a workload cluster and no cp endpoint provided, it won't pass the preflight check. The change here is to inform the api collector that the current task is for a workload cluster.

Reply:
I'm not sure I follow. A workload cluster would still have control plane nodes.  In the case of docker there is no control plan ip specified since it's just handled automatically by docker. We still would want to collect api server logs in that case.

Reply:
This APISever collector will deploy a pod on bootstrap cluster and ping cp IP. From the result of ping, we are able to know the ip is valid or not. 
So even the docker takes over the cp ip, do we still need to run a pod and ping that cp ip?  Except this one, we still can get regular logs.

Reply:
The ping collector isnt needed for the docker provider. Can we just only add it for vsphere?

Reply:
Now it works for vsphere and even other providers as long as the spec contains the cp IP. If no cp host specified for vsphere cluster, the preflight check will filter it out.

Reply:
To add the collector only for vsphere,  it needs to specify the provider=vsphere. There is no such field in cluster-definition.yaml. I just found that the kind of datacenter in the yaml could help us distinguish the providers. We already had a DataCenterConfig function to distribute collectors for vsphere, docker, and cloudstack. Is that ok that we add this collector to vsphere there? 
If we are allowed to do that, there is another problem that we couldnt pass the control plane IP for this collector. emmm, block here.